### ISSUE TITLE
Parameter Question

### ISSUE THREAD

**Issue by ChaJaewook** [oracle/tribuo#48]
How are you?
First of all, thank you for the open source.

I wonder if there is a performance comparison with Python during the practice.

![image](https://user-images.githubusercontent.com/44600628/94408781-3c271c00-01b0-11eb-8015-b818596269d4.png)

Also, I am not sure about the functions of the above parameters in the picture among the Regression parts in the Tutorial.

I would like to ask you what these parameters are actually used in Python and if RNG is the Random Number Generator.

I'm sure you're busy, but I'd appreciate it if you could answer the questions.

Have a good day and thank you again for the good code release. :D

**Comment by Craigacp** [comment]
I'm not sure what Python library you are comparing to? If you mean scikit-learn, then the LinearSGDTrainer for regression models is most similar to this one - https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDRegressor.html. Tribuo's allows you to choose the gradient descent algorithm while scikit-learn's has regularisation support. As Tribuo allows you to supply a gradient optimizer then some of the parameters in the scikit-learn implementation live on Tribuo's optimizers (e.g. learning rate, learning rate schedule, parameter averaging etc). The `epochs` parameter in Tribuo is the same as `max_iters` in scikit-learn. The logging interval is Tribuo specific, the RNG seed is the seed used for the random number generator, which is roughly equivalent to scikit-learn's `random_state` argument.

The parameters are described in the Javadoc for that class - https://tribuo.org/learn/4.0/javadoc/org/tribuo/regression/sgd/linear/LinearSGDTrainer.html#LinearSGDTrainer-org.tribuo.regression.sgd.RegressionObjective-org.tribuo.math.StochasticGradientOptimiser-int-int-int-long- It's always worth checking the Javadoc, it's fairly complete and has references to the academic papers.

**Comment by ChaJaewook** [comment]
Thank you, I want to compare to Tribuo and python library.
So I want to know same parameter value and environment.
For example... Which is faster or accurate... No matter what you use the library in python.
I just know What kind of code you use that when you show on tutorial.

Always appreciate your answer. 

Spelling and grammar may be strange because it is not an English-speaking country.
Although Thank for your answer.
Have a nice day. 😀😀

**Comment by Craigacp** [comment]
There are lots of python libraries which perform linear regression, so without picking a specific one it's difficult to know how to make a useful comparison. Faster is moderately objective (though still hardware dependent), but more accurate will depend on the dataset and hyperparameters chosen. Machine Learning has several "No Free Lunch" theorems which state that it's not possible for one learning algorithm (or hyperparameter set) to be universally better than all others for all datasets.
